packages/model/addon/-private/record-state.ts

```diff
@@ -0,0 +1,392 @@
+import { assert } from '@ember/debug';
+import { dependentKeyCompat } from '@ember/object/compat';
+import { tracked } from '@glimmer/tracking';
+
+import notifyChanges from './notify-changes';
+import cached from './tmp-cached-polyfill';
+
+type RecordData = import('@ember-data/record-data/-private').RecordData;
+type RequestCache = import('@ember-data/store/-private/system/request-cache').default;
+type Model = InstanceType<typeof import('./model')>;
+type StableRecordIdentifier = import('@ember-data/store/-private/ts-interfaces/identifier').StableRecordIdentifier;
+type CoreStore = import('@ember-data/store/-private/system/core-store').default;
+
+function isInvalidError(error) {
+  return error && error.isAdapterError === true && error.code === 'InvalidError';
+}
+
+class Tag {
```

Comment:
Can you please write up a comment describing how/why this is used. Tag is a bit too generic of a thing